module-apps/application-messages/windows/SearchStart.cpp

```diff
@@ -23,12 +23,16 @@ namespace gui
         body = new gui::Item();
         body->setBoundingBox(bodySize());
         addWidget(body);
-        auto text = inputBox(this, utils::translate("common_search_uc"), "search_32px_W_G");
+
+        const auto text = inputBox(this, utils::translate("common_search_uc"), "search_32px_W_G");
         text->setInputMode(new InputMode(
             {InputMode::Abc, InputMode::ABC, InputMode::abc, InputMode::digit},
             [=](const UTF8 &Text) { application->getCurrentWindow()->navBarTemporaryMode(Text); },
             [=]() { application->getCurrentWindow()->navBarRestoreFromTemporaryMode(); },
-            [=]() { application->getCurrentWindow()->selectSpecialCharacter(); }));
+            [=]() { application->getCurrentWindow()->selectSpecialCharacter(); },
+            [=](std::function<void()> restoreFunction) {
+                application->getCurrentWindow()->startInputModeRestoreTimer(std::move(restoreFunction));
```

Comment:
Maybe this could be put to another file and not repeated? This is de facto a copy-paste in several places

Reply:
That's true, but it's exactly the same with all the lambda functions passed to the InputMode ctor - I just followed the same pattern that someone else designed before, to avoid doing some bigger refactor.